compiler/rustc_const_eval/src/interpret/cast.rs

```diff
@@ -128,23 +141,6 @@ impl<'mir, 'tcx: 'mir, M: Machine<'mir, 'tcx>> InterpCx<'mir, 'tcx, M> {
 
         // # First handle non-scalar source values.
 
-        // Handle cast from a ZST enum (0 or 1 variants).
-        match src.layout.variants {
-            Variants::Single { index } => {
-                if src.layout.abi.is_uninhabited() {
-                    // This is dead code, because an uninhabited enum is UB to
-                    // instantiate.
-                    throw_ub!(Unreachable);
-                }
-                if let Some(discr) = src.layout.ty.discriminant_for_variant(*self.tcx, index) {
-                    assert!(src.layout.is_zst());
-                    let discr_layout = self.layout_of(discr.ty)?;
-                    return Ok(self.cast_from_int_like(discr.val, discr_layout, cast_ty).into());
-                }
-            }
-            Variants::Multiple { .. } => {}
-        }
-
```

Comment:
Note to self: `misc_cast` ends in a `cast_from_int_like` call, which an `enum` `src` would fall into, prompting an ICE as it's not an integer-like primitive type (so no accidents should be possible).

Reply:
enums are "integer-like" if they have `Scalar` ABI.